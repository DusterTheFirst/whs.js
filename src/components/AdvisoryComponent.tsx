/*!
 * Copyright (C) 2018  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { StyleSheet, Text, View } from "react-native";

interface IProps {
    teacher: string;
    room: number;
}
export default function AdvisoryComponent({teacher, room}: IProps) {
    return (
        <View style={styles.container}>
            <View style={styles.dualView}>
                <Text style={styles.title}>Advisory</Text>
                <Text style={styles.dim}>9:38 - 9:46 AM</Text>
            </View>
            <View style={[styles.dualView, styles.info]}>
                <Text style={[styles.dim, styles.teacher]} numberOfLines={1}>{teacher}</Text>
                <Text style={[styles.dim, styles.room]} numberOfLines={1}>Room {room}</Text>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: "#FFFFFF",
        // borderBottomColor: "#CECECE",
        // borderBottomWidth: 1,
        height: 75,
        padding: 10,
        width: "100%"
    },
    dim: {
        color: "#808080"
    },
    dualView: {
        alignItems: "baseline",
        flexDirection: "row",
        flexWrap: "nowrap",
        justifyContent: "space-between"
    },
    info: {
        marginTop: 5
    },
    room: {
        overflow: "visible"
    },
    teacher: {
        maxWidth: "50%"
    },
    title: {
        color: "#A0A0A0",
        fontSize: 25,
        fontWeight: "bold"
    }
});