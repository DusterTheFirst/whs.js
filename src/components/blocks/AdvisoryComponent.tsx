/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { StyleSheet, Text, View } from "react-native";

const styles = StyleSheet.create({
    container: {
        backgroundColor: "#FFFFFF",
        // borderBottomColor: "#CECECE",
        // borderBottomWidth: 1,
        height: 75,
        padding: 10,
        width: "100%"
    },
    dim: {
        color: "#808080"
    },
    dualView: {
        alignItems: "baseline",
        flexDirection: "row",
        flexWrap: "nowrap",
        justifyContent: "space-between"
    },
    info: {
        marginTop: 5
    },
    room: {
        overflow: "visible"
    },
    teacher: {
        maxWidth: "50%"
    },
    title: {
        color: "#A0A0A0",
        fontSize: 25,
        fontWeight: "bold"
    }
});

export default function AdvisoryComponent({teacher, room}: IAdvisory) {
    return (
        <View style={styles.container}>
            <View style={styles.dualView}>
                <Text style={styles.title}>Advisory</Text>
                <Text style={styles.dim}>9:38 - 9:46 AM</Text>
            </View>
            <View style={[styles.dualView, styles.info]}>
                <Text style={[styles.dim, styles.teacher]} numberOfLines={1}>{teacher}</Text>
                <Text style={[styles.dim, styles.room]} numberOfLines={1}>Room {room}</Text>
            </View>
        </View>
    );
}