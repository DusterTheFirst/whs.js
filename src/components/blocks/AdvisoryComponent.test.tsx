/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { render } from "react-native-testing-library";
import AdvisoryComponent from "./AdvisoryComponent";

describe("Tests different variations of <AdvisoryComponent>", () => {
    it("Renders AdvisoryComponent with a string room", () => {
        let comp = render(<AdvisoryComponent room="Office" teacher="Mr. Teacher Man"/>);

        expect(comp.toJSON()).toMatchSnapshot();
    });
    it("Renders AdvisoryComponent with a numerical room", () => {
        let comp = render(<AdvisoryComponent room={132} teacher="Mr. Teacher Man"/>);

        expect(comp.toJSON()).toMatchSnapshot();
    });
});