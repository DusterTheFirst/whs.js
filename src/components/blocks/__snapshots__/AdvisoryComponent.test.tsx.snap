// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests different variations of <AdvisoryComponent> Renders AdvisoryComponent with a numerical room 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      ellipsizeMode="middle"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#444444",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 25,
            "fontWeight": "bold",
            "textAlign": "left",
          },
        ]
      }
    >
      Advisory
    </Text>
    <View
      flex={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#808080",
              "paddingLeft": 5,
              "textAlign": "right",
            },
          ]
        }
      >
        11:00
         - 
        11:30 PM
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "space-between",
          "marginTop": 5,
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#808080",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "textAlign": "left",
          },
        ]
      }
    >
      Mr. Teacher Man
    </Text>
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#808080",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "textAlign": "right",
          },
        ]
      }
    >
      Room 132
    </Text>
  </View>
</View>
`;

exports[`Tests different variations of <AdvisoryComponent> Renders AdvisoryComponent with a string room 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      ellipsizeMode="middle"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#444444",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 25,
            "fontWeight": "bold",
            "textAlign": "left",
          },
        ]
      }
    >
      Advisory
    </Text>
    <View
      flex={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#808080",
              "paddingLeft": 5,
              "textAlign": "right",
            },
          ]
        }
      >
        9:38
         - 
        9:46 AM
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "space-between",
          "marginTop": 5,
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#808080",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "textAlign": "left",
          },
        ]
      }
    >
      Mr. Teacher Man
    </Text>
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#808080",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "textAlign": "right",
          },
        ]
      }
    >
      Office
    </Text>
  </View>
</View>
`;
